jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -55,7 +57,6 @@
  * a file, a URL or an entry in a jar file.
  * </p>
  */
-//TODO remove
 public abstract class Resource implements ResourceFactory, Closeable
```

Comment:
Should we implement addPath and getResource in terms of URI operations to define exactly how they should work for all Resources. Everything else is just an optimization (except for the ResourceCollection)

Reply:
The only `getResource()` implementation actually delegates to `addPath()` so at least one of the two methods should go.

As we discussed yesterday, a single `Resource resolve(URI subUri)` method should replace them both if possible.

Reply:
+1 for a single method.

But I'd still like to see that implemented in the base class in terms of URIs, so that the semantic is made clear (even if not efficient).

Reply:
With some some refactoring of `Resource` and its subclasses, I think I've managed to generalize the resolve logic and implement it in the abstract `Resource` class.